kafka/producer.go

```diff
@@ -0,0 +1,70 @@
+package kafka
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+type Producer interface {
+	Send([]byte) error
+}
+
+// Producer Implementation
+type KafkaProducer struct {
+	producer sarama.SyncProducer
+	topic    string
+	ctx      *ProducerManager
+}
+
+func NewProducer(ctx *ProducerManager, topic string, producer sarama.SyncProducer) Producer {
+	return &KafkaProducer{
+		producer: producer,
+		topic:    topic,
+		ctx:      ctx,
+	}
+}
+
+func (self *KafkaProducer) Send(payload []byte) error {
+	_, _, err := self.producer.SendMessage(&sarama.ProducerMessage{
+		Topic: self.topic,
+		Value: sarama.ByteEncoder(payload),
+	})
+
+	if err != nil {
+		if err == sarama.ErrBrokerNotAvailable || err == sarama.ErrClosedClient {
+			// Signal We should reconnect
+			self.ctx.SignalReconnect()
+		}
+		return err
+	}
+	return nil
+}
+
+func (self *KafkaProducer) Get(payload []byte) error {
```

Comment:
The ``Get`` method is identical to ``Send`` method above. Is it supposed to be so? Maybe call one from the other then or have a third, helper method, both will be calling to avoid code duplication?